packages/core/src/lib/callback-handler.ts

```diff
@@ -154,6 +154,13 @@ export async function handleLogin(
 
       return { session, user: userByAccount, isNewUser }
     } else {
+      const { provider } = options as InternalOptions<"oauth" | "oidc">
+      account = Object.assign(provider.account({ ...account }), {
```

Comment:
Since we're going to override the `providerAccountId`, `provider`, and `type` anyway, do you think we should exclude these fields from the `account` callback?
How about the `userId` field, should we allow them to override?

Reply:
Good point! I was thinking about that... I guess we could start with a stricter API and if a need arises, we could add it back then.